File: jobs/transform/src/transform/settings_db.py
"""Load app settings from the kanjo.app_settings table."""

import psycopg


def load_settings_from_db(database_url: str) -> dict[str, str]:
    """Query kanjo.app_settings and return non-empty values as a dict."""
    settings: dict[str, str] = {}
    try:
        with psycopg.connect(database_url) as conn:
            with conn.cursor() as cur:
                cur.execute(
                    "SELECT key, value FROM kanjo.app_settings WHERE value != ''"
                )
                for row in cur.fetchall():
                    settings[row[0]] = row[1]
    except Exception:
        pass
    return settings

File: jobs/scraper/src/settings-db.ts
import type pg from 'pg';

interface AppSetting {
  key: string;
  value: string;
}

/**
 * Load app settings from the kanjo.app_settings table.
 * Returns a map of key -> value for non-empty settings.
 */
export async function loadSettingsFromDb(pool: pg.Pool): Promise<Map<string, string>> {
  const result = await pool.query<AppSetting>(
    `SELECT key, value FROM kanjo.app_settings WHERE value != ''`
  );

  const settings = new Map<string, string>();
  for (const row of result.rows) {
    settings.set(row.key, row.value);
  }
  return settings;
}
